src/PhpDoc/Tag/TypeAliasImportTag.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\PhpDoc\Tag;
+
+final class TypeAliasImportTag
+{
+
+	private string $importedAlias;
+
+	private \PHPStan\Type\Type $importedFrom;
```

Comment:
One idea about this - I think that importedFrom can only ever be IdentifierTypeNode, isn't PHPStan\Type\Type unnecesarily wide type here? It'd be useful if this was just a string with the class name where the tag is from?

Reply:
I think it would still need to be resolved in the right `NameScope`, the only difference is that it would have to be done later. Currently it happens [early](https://github.com/phpstan/phpstan-src/pull/460/files#diff-649aa72d6962638843a5b8abd489350f92014f519ae86cb9efd388e4ed3dd967R398) and both the [resolver](https://github.com/phpstan/phpstan-src/pull/460/files#diff-1cb8c0f16a3848b23fbd09ac0ce1f5ef06d9e8365d2aded0c6028b8035c78c6fR786) and the [rule](https://github.com/phpstan/phpstan-src/pull/460/files#diff-0d1d8d1217a959a18b46ff0e0b1e1696576e49cd2112fe3b4ac1ecc1d1f56cecR80) check if it's an actual `ObjectType` because it [might not be](https://github.com/phpstan/phpstan-src/pull/460/files#diff-f5beb741d2711011522821608011df3f7ffa1d825db7996eab628378ee24724fR60).

Reply:
We already eliminated that situation here, right? https://github.com/phpstan/phpdoc-parser/pull/70 so it should be a parse error from the phpdoc-parser

Reply:
You can still `import from int`, it's a valid identifier so the parser is ok with it, but it's not a valid class name and resolves to `IntegerType`. Which, I agree, is weird. Your suggestion makes sense, we could treat it as a class name and then fail because class `WhateverTheScopeIs\int` does not exist.

Reply:
I see, `int` is still a valid identifier and the difference is made in phpstan/phpstan in TypeNodeResolver. So parser doesn't know about this.

Yeah, i think it would actually be nicer to have `string` instead of `PHPStan\Type\Type` in TypeAliasImportTag, and call just `NameScope::resolveStringName()` instead of the whole `TypeNodeResolver`. Having to do `instanceof ObjectType` doesn't sit well with me because `TypeAliasImportTag` with IntegerType never makes sense, it shouldn't be created in the first place.